exotica_core/src/problems/dynamic_time_indexed_shooting_problem.cpp

```diff
@@ -272,6 +272,16 @@ void DynamicTimeIndexedShootingProblem::ReinitializeVariables()
     }
     cost.ReinitializeVariables(T_, shared_from_this(), cost_Phi);
 
+    if (this->parameters_.WarmStartWithInverseDynamics)
```

Comment:
Should this be in reinitialize or also in PreUpdate? What if someone changes X after setting the T (which calles Reinitialize)

Reply:
Duplicate it, you mean? I don't really see the use-case. Right now this is called in the constructor after everything else.

Reply:
Yes and that is the problem! If it is only in the constructor and not in PreUpdate, it will be ignored if you change `T` or anything else after construction...

Reply:
Ok, so why not put it just in `PreUpdate` then?

Reply:
Yep, please do - and check that your solvers are calling it at the beginning of Solve()